### PR TITLE
Field updating

### PR DESCRIPTION
**Pull request by jcontonio**
* Adds in request and receive fields actions/reducers
* Adds DocumentFieldsContainer that does the updating of child fields
* Adds TextComponent with some comments on the React Hooks

To Test: 

[This document](http://localhost:3000/d/project/110000/document/120001) is the only one with fields. You should be able to edit the textarea and write markdown

### CODE REVIEW

File: api/src/constants/fieldTypes.js
Comment (jjidt) [inline api/src/constants/fieldTypes.js:2]: I don't think we'll be able to share this since these are separate docker containers, unless we publish the constants file to a private npm module, or import it from github for both services, but I could be wrong

File: web/src/client/js/actions/field.js
Comment (jjidt) [inline web/src/client/js/actions/field.js]: you changed id to fieldId above, do you want to do that here as well for consistency?

File: web/src/client/js/components/Document/DocumentFieldsContainer.js
Comment (jjidt) [inline web/src/client/js/components/Document/DocumentFieldsContainer.js]: do we still need this logging?

File: web/src/client/js/components/Document/DocumentFieldsContainer.js
Comment (jjidt) [inline web/src/client/js/components/Document/DocumentFieldsContainer.js:21]: or this?

**Review by jjidt — DISMISSED** [review]
Really cool to see this working, code looks good overall, a few comments

@dheniges seems like there could potentially be some edge cases here, can you also click around whenever you get a chance

File: api/src/constants/fieldTypes.js
Comment (jcontonio) [reply api/src/constants/fieldTypes.js:2]: Oh right...

File: web/src/client/js/components/Document/DocumentFieldsContainer.js
Comment (jcontonio) [reply web/src/client/js/components/Document/DocumentFieldsContainer.js:21]: This one I left in on purpose, because of tricky useEffect sideeffects. I want to make sure that if we end up doing something that hits the API multiple times (different hook dependencies did this), that we can see it for a while. Purely just for dev

File: api/src/controllers/documentFields.js
Comment (dheniges) [inline api/src/controllers/documentFields.js]: this comment is inaccurate, the body schema (aka the payload to this endpoint) should definitely be validated. The problem is the body schema between the "create" and "update" endpoints should not the same (but currently is, as implemented). I updated this Asana task to reflect the update: https://app.asana.com/0/1103604360526176/1118485962871954

For now, let's update the comment.
```suggestion
    // validateBody(bodySchema), // TODO: update body schema for PUT request. All fields should not be required on PUT
```

File: docs/data.md
Comment (dheniges) [inline docs/data.md]: This README update isn't accurate in all scenarios. You can't always undo migrations (switching branches between PRs with different migrations for instance, or not being fully migrated, etc.).

You ran into problems with `rebuilddev` because you'd already run `cleardev`, which is why the doc lines you deleted said "OR".

The clear and rebuild commands also do a complete rebuild of the docker images, which the migration commands don't help with at all. The migration commands are also documented right above this in the "Migrations" section.

So I'm not quite sure what doc updates would help here, or what should be clarified?

File: web/src/client/js/components/Document/DocumentFieldsContainer.js
Comment (dheniges) [inline web/src/client/js/components/Document/DocumentFieldsContainer.js]: Since this isn't using mapStateToProps, this component doesn't need the `connect()` middleware. You can directly import the store and call `Store.dispatch(action)`

File: web/src/client/js/components/Document/DocumentContainer.js
Comment (dheniges) [inline web/src/client/js/components/Document/DocumentContainer.js]: Shouldn't this be in the `DocumentFieldsContainer`? If avoidable, it's best not to pass props to containers, and I don't see the fields being used in this component or the DocumentComponent. The fields data just gets passed through several layers as props.

**Review by dheniges — DISMISSED** [review]
The functionality seems great, I didn't hit any issues playing around with that text field 💪 

A couple comments on comments/docs, and some simplifications in react components

File: web/src/client/js/components/Document/DocumentFieldsContainer.js
Comment (jcontonio) [reply web/src/client/js/components/Document/DocumentFieldsContainer.js]: If I switch to DocumentFieldsContaainer grabbing all the fields from state, then this is a non-issue, yeah?

Comment (dheniges) [reply web/src/client/js/components/Document/DocumentFieldsContainer.js]: you'll still want to use the `useDataService` hook to get the data, not the connect() mapStateToProps. Otherwise you won't trigger the action to fetch the fields from the API

**Comment by jcontonio** [comment]
This is ready to go, but hold off until JJ's endpoint PR gets merged in, due to casting and docId / id in params

File: docs/data.md
Comment (dheniges) [reply docs/data.md]: @jcontonio this is still a thing ^. I don't want to remove the docs on `cleardev` or `rebuilddev`, limited as they are. The migration stuff commands are documented in this file already, in the previous section.

So, revert this change? Or... whatever you want to do except delete the docs on cleardev/rebuilddev

Comment (jcontonio) [reply docs/data.md]: Reverted, should be all set

**Review by dheniges — APPROVED** [review]
so hawt 🔥

**Pull request merged**
